File: go/src/utils/string_utils.go
package utils

func ReverseString(s string) string {
	n := len(s)
	r := []rune(s)

	for i := 0; i < n/2; i++ {
		r[i], r[n-i-1] = r[n-i-1], r[i]
	}
	return string(r)
}

func ASCIISum(s string) int {
	sum := 0
	for _, v := range s {
		sum += int(v)
	}
	return sum
}

func WordScore(s string) int {
	return ASCIISum(s) - len(s)*64
}

File: python/utils/string_utils.py


def word_score(word):
    """
    Returns the sum of the ascii values of the letters composing the word.
    Only works on uppercase words.
    """
    # 64 = ord('A')
    total = 0
    for letter in word:
        total += ord(letter) - 64

    return total


def ascii_sum(data):
    total = 0
    for char in data:
        total += ord(char)

    return total
